### PR TITLE
feat: make locales reusable & add zh-Hant locale

### PR DESCRIPTION
**Pull request by CoolPlayLin**

### CODE REVIEW

**Comment by cexbrayat** [comment]
@btea Can you take a look please? I sadly can't read chinese ^^ It looks very similar to zh-CN.json but maybe there are subtleties I don't understand

File: locales/zh-TW.json [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+  },
+  "needsPinia": {
+    "message": "是否引進 Pinia 用於狀態管理？"
+  },
+  "needsVitest": {
+    "message": "是否引入 Vitest 用於單元測試"
+  },
+  "needsE2eTesting": {
+    "message": "是否要引入一款端對端（End to End）測試工具？",
+    "selectOptions": {
```
Comment (cexbrayat) [inline locales/zh-TW.json:33]: There is a `hint` key here:
```json
"hint": "- Use arrow-keys. Return to submit.",
```

**Review by cexbrayat — CHANGES_REQUESTED** [review]

**Comment by CoolPlayLin** [comment]
> @btea Can you take a look please? I sadly can't read chinese ^^ It looks very similar to zh-CN.json but maybe there are subtleties I don't understand

Hi @cexbrayat 

One you mentioned is used in China mainland and this file is used in Taiwan. Although they are both Chinese, they have something different. Chinese in China mainland was simplified in order to get education popularized. But in Hong Kong Macao and Taiwan, they still use the traditional Chinese character.

File: locales/zh-TW.json [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+  },
+  "errors": {
+    "operationCancelled": "操作取消"
+  },
+  "defaultToggleOptions": {
+    "active": "是",
+    "inactive": "否"
+  },
+  "infos": {
+    "scaffolding": "正在建構專案",
```
Comment (btea) [inline locales/zh-TW.json:64]: @CoolPlayLin 你好！请问这里为什么使用 `專案` 而不是像第三行一樣使用 `項目` 呢？

**Review by btea — APPROVED** [review]
LGTM

File: locales/zh-TW.json (hunk 1 above)
Comment (CoolPlayLin) [reply locales/zh-TW.json:34]: It appears this Change Request still prevent the merging. Could you please resolve this?

**Comment by CoolPlayLin** [comment]
@btea 是不是有必要再增加`zh-HK` `zh-MO` `zh-SG`的本地化文件呢？

**Comment by btea** [comment]
@CoolPlayLin 我不确定 `getLocale` 是否有这么细致的划分，你能确定吗？

**Comment by CoolPlayLin** [comment]
@btea 我尝试将系统切换成其他语言查看`Intl.DateTimeFormat().resolvedOptions().locale`获取的值是否有划分

**Comment by CoolPlayLin** [comment]
Anyway, it returned the expected result after switching language on Windows
image

**Comment by CoolPlayLin** [comment]
Another PR opened in #375

**Comment by CoolPlayLin** [comment]
@btea 我能想到的只有这个方案了，不知道是否可行

File: utils/getLanguage.ts [hunk 3]
```diff
@@ -52,7 +83,7 @@ function getLocale() {
     // TODO: Windows support if needed, could consider https://www.npmjs.com/package/os-locale
     'en-US' // Default fallback
 
-  const locale = shellLocale.split('.')[0].replace('_', '-')
+  const locale = linkLocale(shellLocale.split('.')[0].replace('_', '-'))
```
Comment (btea) [inline utils/getLanguage.ts:86]:
```suggestion
  return linkLocale(shellLocale.split('.')[0].replace('_', '-'))
```

File: utils/getLanguage.ts [hunk 4]
```diff
@@ -43,6 +43,37 @@ interface Language {
...
+      linkedLocale = 'zh-Hans'
+      break
+    case 'zh-SG':
+      linkedLocale = 'zh-Hans'
+      break
+    default:
+      linkedLocale = locale
+  }
+
+  return linkLocale
```
Comment (btea) [inline utils/getLanguage.ts:74]:
```suggestion
  return linkedLocale
```

File: utils/getLanguage.ts [hunk 5]
```diff
@@ -43,6 +43,37 @@ interface Language {
...
+  let linkedLocale: string
+  switch (locale) {
+    case 'zh-TW':
+      linkedLocale = 'zh-Hant'
+      break
+    case 'zh-HK':
+      linkedLocale = 'zh-Hant'
+      break
+    case 'zh-MO':
+      linkedLocale = 'zh-Hant'
```
Comment (btea) [inline utils/getLanguage.ts:63]:
```suggestion
    case 'zh-TW':
    case 'zh-HK':
    case 'zh-MO':
      linkedLocale = 'zh-Hant'
      break
```

**Review by btea — COMMENTED** [review]
@CoolPlayLin  I've added some suggestions for modifications, you can take a look.

**Review by btea — APPROVED** [review]

**Comment by CoolPlayLin** [comment]
@cexbrayat

File: locales/zh-Hant.json [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+  "errors": {
+    "operationCancelled": "操作取消"
+  },
+  "defaultToggleOptions": {
+    "active": "是",
+    "inactive": "否"
+  },
+  "infos": {
+    "scaffolding": "正在建構項目",
+    "done": "項目建置完成，可執行以下命令："
```
Comment (sodatea) [inline locales/zh-Hant.json:66]: 感覺應該也是「專案建置」而不是「項目建置」

**Review by sodatea — CHANGES_REQUESTED** [review]
我不太熟悉港台的相關術語，@johnsoncodehk @antfu 能否麻煩幫忙審閱一下？這兩種 locale 能合併嗎？

**Comment by CoolPlayLin** [comment]
@antfu @johnsoncodehk 能否麻煩查看一下，謝謝！

File: locales/zh-Hant.json [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+  },
+  "needsPinia": {
+    "message": "是否引入 Pinia 用於狀態管理？"
+  },
+  "needsVitest": {
+    "message": "是否引入 Vitest 用於單元測試"
+  },
+  "needsE2eTesting": {
+    "message": "是否要引入一款端對端（End to End）測試工具？",
+    "hint": "- 使用箭頭切換按Enter確認。",
```
Comment (antfu) [inline locales/zh-Hant.json:33]:
```suggestion
    "hint": "- 使用箭頭切換按 Enter 確認。",
```

File: locales/zh-Hant.json [hunk 8]
```diff
@@ -0,0 +1,68 @@
+{
+  "projectName": {
+    "message": "請輸入專案名稱："
+  },
+  "shouldOverwrite": {
+    "dirForPrompts": {
+      "current": "目前目錄：",
+      "target": "目標目錄："
```
Comment (antfu) [inline locales/zh-Hant.json:8]:
```suggestion
      "current": "目前資料夾：",
      "target": "目標資料夾："
```

**Review by antfu — APPROVED** [review]
我了解的台湾地区的术语有限，但 Overall looks good

**Comment by CoolPlayLin** [comment]
Any update?@cexbrayat @sodatea

**Comment by cexbrayat** [comment]
I'll let @sodatea check this one, as I can't help with these locales

**Comment by sodatea** [comment]
We still need someone familiar with Hong Kong technology terms to review this PR.
I don't feel confident about merging these locales.

**Comment by CoolPlayLin** [comment]
@johnsoncodehk 可以幫忙看一下嗎?

**Comment by CoolPlayLin** [comment]
>We still need someone familiar with Hong Kong technology terms to review this PR.

Could someone take a look? Thanks!

CC @ygj6 @yoyo930021

**Review by yoyo930021 — COMMENTED** [review]
Perhaps some words would be better

**Review by yoyo930021 — APPROVED** [review]
LGTM

**Review by sodatea — APPROVED** [review]

**Comment by sodatea** [comment]
We still haven't got a review on the `zh-HK` locale.
But I think we need to move forward now, and we can revert when there are real user complaints.

Thanks for the contribution and patience!

**Pull request merged**